samza-api/src/main/java/org/apache/samza/system/StreamSpec.java

```diff
@@ -143,13 +145,17 @@ public StreamSpec(String id, String physicalName, String systemName, int partiti
       throw new NullPointerException("Parameter 'systemName' must not be null");
     }
 
+    if (partitionCount < 1) {
+      throw new NullPointerException("Parameter 'partitionCount' must not be greater than 0");
```

Comment:
Two comments on this one:
- Shouldn't this be IllegalArgumentException?
- "Parameter 'partitionCount' must be greater than 0"